### PR TITLE
[TIMOB-23612] Android: Improve requestPerm docs

### PR DESCRIPTION
**Pull request by ashcoding**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23612
Android: Improve `Ti.Android.requestPermissions()` documentation

**Optional Description:**

This PR is to improve the documentation of
Ti.Android.requestPermissions() documentation.

### CODE REVIEW

File: apidoc/Titanium/Android/Android.yml [hunk 1]
```diff
@@ -93,6 +93,36 @@ description: |
     For more information about services, see the
     [Titanium.Android.Service API reference](Titanium.Android.Service).
 
+    #### Permissions
+
+    Starting from Android 6.0 (API level 23), users need to grant permissions to apps while the
```
Comment (hansemannn) [inline apidoc/Titanium/Android/Android.yml]: Not all permissions need to be granted while the app is running, right?

File: apidoc/Titanium/Android/Android.yml [hunk 2]
```diff
@@ -93,6 +93,36 @@ description: |
...
+    app is running. You can read it more [here](https://developer.android.com/training/permissions/requesting.html).
+
+    In Titanium SDK, to support this, we have the method `Ti.Android.requestPermissions` starting
+    from Titanium SDK 5.4.0. This is used to requests any permission you may need. An example of
+    using this is shown below.
+
+        var cameraPermission = "android.permission.CAMERA";
+        var storagePermission = "android.permission.READ_EXTERNAL_STORAGE";
+        var hasCameraPerm = Ti.Android.hasPermission(cameraPermission);
+        var hasStoragePerm = Ti.Android.hasPermission(storagePermission);
```
Comment (hansemannn) [inline apidoc/Titanium/Android/Android.yml]: Naming: `hasCameraPerm` => `hasCameraPermissions`, same for storage.

**Comment by hansemannn** [comment]
Left two 2 minor comments for improved readability. Should we link to this section from the API's that require these permissions (e.g. Ti.Contacts) or do people know about this section? I would personally not think about primary searching there, but in the description of the related API's.

**Comment by ashcoding** [comment]
That's a good point. I'll leave additional notes in those section (eg Ti.Contacts, Ti.Geolocation) mentioning about this general permission API. What do you think?

**Comment by hansemannn** [comment]
That would be awesome! So we have everything linked with each other and people don't get confused. Think of the API's that might need the note and we're good to go. Thanks man! 🚀

**Comment by ashcoding** [comment]
Comments addressed.

**Comment by ashcoding** [comment]
Comments addressed.

**Comment by hansemannn** [comment]
Love it! Approved 🚀

**Pull request merged**